Rocket.Chat/Models/File.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  File.swift
+//  Rocket.Chat
+//
+//  Created by Filipe Alvarenga on 11/04/18.
+//  Copyright © 2018 Rocket.Chat. All rights reserved.
+//
+
+import Foundation
+import RealmSwift
+
+class File: BaseModel {
+    @objc dynamic var name = ""
+    @objc dynamic var rid = ""
+    @objc dynamic var size: Double = 0
+    @objc dynamic var type = ""
+    @objc dynamic var fileDescription = ""
+    @objc dynamic var store = ""
+    @objc dynamic var isComplete = false
+    @objc dynamic var isUploading = false
+    @objc dynamic var fileExtension = ""
+    @objc dynamic var progress: Int = 0
+    @objc dynamic var user: User?
+    @objc dynamic var uploadedAt: Date?
+    @objc dynamic var url = ""
+}
+
+extension File {
+    var username: String {
+        return user?.username ?? ""
+    }
+
+    var isImage: Bool {
+        let imageTypes = ["gif", "jpg", "png", "jpeg", "image"]
+        return imageTypes.compactMap { type.range(of: $0, options: .caseInsensitive) != nil ? true : nil }.first ?? false
+    }
+
+    var isGif: Bool {
+        let gifTypes = ["gif"]
+        return gifTypes.compactMap { type.range(of: $0, options: .caseInsensitive) != nil ? true : nil }.first ?? false
+    }
+
+    var isVideo: Bool {
+        let videoTypes = ["mp4", "mov", "video", "flv"]
+        return videoTypes.compactMap { type.range(of: $0, options: .caseInsensitive) != nil ? true : nil }.first ?? false
+    }
+
+    var isAudio: Bool {
+        let audioTypes = ["mp3", "wav", "3gp", "audio"]
+        return audioTypes.compactMap { type.range(of: $0, options: .caseInsensitive) != nil ? true : nil }.first ?? false
+    }
+
```

Comment:
What if the file is audio mp4?
why not just `return type.range(of: "audio", options: .caseInsensitive) != nil` ?

The same thing for video and image.

Reply:
Makes sense! 👍

Reply:
@mrqbmrqb Made the changes here: https://github.com/RocketChat/Rocket.Chat.iOS/pull/1663.